test/suite/test_backup_base.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2021 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+import os
+import wttest
+from suite_subprocess import suite_subprocess
+from helper import compare_files
+
+# Shared base class used by backup tests.
+class test_backup_base(wttest.WiredTigerTestCase, suite_subprocess):
+    #
+    # Set up all the directories needed for the test. We have a full backup directory for each
+    # iteration and an incremental backup for each iteration. That way we can compare the full and
+    # incremental each time through.
+    #
+    def setup_directories(self, max_iteration, home_incr, home_full, logpath):
+        for i in range(0, max_iteration):
+            # The log directory is a subdirectory of the home directory,
+            # creating that will make the home directory also.
+
+            home_incr_dir = home_incr + '.' + str(i)
+            if os.path.exists(home_incr_dir):
+                os.remove(home_incr_dir)
+            os.makedirs(home_incr_dir + '/' + logpath)
+
+            if i == 0:
+                continue
+            home_full_dir = home_full + '.' + str(i)
+            if os.path.exists(home_full_dir):
+                os.remove(home_full_dir)
+            os.makedirs(home_full_dir + '/' + logpath)
+    #
+    # Compare against two directory paths using the wt dump command. If the second directory path is not provided
+    # it checks if the compare
+    #
+    def compare_backups(self, uri, dir1, file1, run_recovery = False, dir2 = None, file2 = None):
```

Comment:
Why do we need output file names as args? As far as I can tell the tests don't use them anywhere else and don't care about them. Only this function uses them to compare the two directories so I don't think they need to be args and the function can just make its own file names.

Reply:
the outfiles are useful when you need to perserve the files if you wanted to look at the test directory for further information. Since the outfiles are created in the base test directory, test files like test_backup14.py, test_backup15.py create different outfiles, such that you can visualise the different outfiles that are made by the test. If we wanna remove the output arguments, we can still maintain the backups by placing them into their individual directories instead of the base directory. What opinions do you have?

Reply:
Update: talked with tammy, removed the outfiles arguments, with new version of **compare_backups**, now we pass in a "suffix" to account for the different possible outfiles